aderyn_core/src/detect/nc/useless_internal_function.rs

```diff
@@ -0,0 +1,92 @@
+use std::{collections::BTreeMap, error::Error};
+
+use crate::{
+    ast::{NodeID, Visibility},
+    capture,
+    context::workspace_context::WorkspaceContext,
+    detect::{
+        detector::{IssueDetector, IssueDetectorNamePool, IssueSeverity, ReusableDetector},
+        reusable::IdentifiersThatReferenceAFunctionDetector,
+    },
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct UselessInternalFunctionDetector {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for UselessInternalFunctionDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        let internal_functions = context
+            .function_definitions()
+            .into_iter()
+            .filter(|&function| matches!(function.visibility, Visibility::Internal));
+
+        for internal_function in internal_functions {
+            if IdentifiersThatReferenceAFunctionDetector::default()
```

Comment:
You used the reusable detector 🚀 

I'm going to approve this as-is, but I remember you were trying out a `Helper` pattern instead of this reusable detector pattern. Do you think that would make this cleaner?

Reply:
Yes ! Let me try it : )